### DISCUSSION TITLE
embedded-SQL C code

### DISCUSSION THREAD

**Discussion by pbderr** [microsoft/vscode-cpptools#8054] [Q&A]
My project has lots of C code with embedded SQL processed by the Oracle Pro*C precompiler. (https://en.wikipedia.org/wiki/Pro*C)
The SQL code is prefixed by "exec sql" (case insensitive) and terminated by a semi-colon (;), possibly multiple lines.

For example:
    exec sql select count(*)
               into :PayerCnt
               from t_re_pat_liab
              where sak_recip = :sakRecip;

How can I configure the VScode C/C++ extension to not consider this to be an error?

My SQL-C source files are named \*.sc and in settings.json I can make them be recognized as C code like this:
  "files.associations": {
        "*.sc": "c"
}

**Comment by sean-mcmanus** [comment]
You could wrap your SQL code in
```
#ifndef __INTELLISENSE__
#endif
```

The other approach is get Oracle's Pro*C compiler developers to contribute changes to the edg compiler we use (https://www.edg.com/c), similar to how we integrated support for the CUDA language extensions by NVIDIA. However, our compiler team would have to agree to hook up the necessary changes.

**Comment by pbderr** [comment]
Well, the "`#ifndef __INTELLISENSE__`" solution works (and thanks for pointing that out), but it is not really practical since we have about 900 of these source modules with many thousands of embedded SQL statements.   I was hoping for some option that would allow VScode to ignore certain statements based on some regular expression.
Getting Oracle's Pro*C compiler developers to contribute changes seems like a long shot, but could be pursued.
I'm looking for a solution to this issue because I'm trying to promote the use of VScode by our C developers.

**Reply by sean-mcmanus** [reply]
I've added a feature request at https://github.com/microsoft/vscode-cpptools/issues/8069, but it'd probably need more upvotes. Our parser doesn't have the ability to use a regex to ignore certain expressions, so that would have to be added.
